spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactoryTests.java

```diff
@@ -212,4 +216,21 @@ public void testCreatesConnectionWithGivenFactory() {
 		assertThat(mockConnectionFactory.getThreadFactory()).isEqualTo(connectionThreadFactory);
 	}
 
+	@Test
+	public void testConnectionCreatingBackOff() throws Exception {
+		int maxAttempts = 2;
+		long interval = 500L;
+		com.rabbitmq.client.Connection mockConnection = mock(com.rabbitmq.client.Connection.class);
+		given(mockConnection.createChannel()).willReturn(null);
+		SimpleConnection simpleConnection = new SimpleConnection(mockConnection, 5,
+				new FixedBackOff(interval, maxAttempts).start());
+		StopWatch stopWatch = new StopWatch();
+		stopWatch.start();
+		assertThatExceptionOfType(AmqpResourceNotAvailableException.class).isThrownBy(() -> {
+			simpleConnection.createChannel(false);
+		});
+		stopWatch.stop();
+		assertThat(stopWatch.getTotalTimeMillis()).isGreaterThanOrEqualTo(maxAttempts * interval);
```

Comment:
I wonder if we really need to wait in this test the whole `1 second`.
There are at about 2000 tests in the project.
Now imaging every single would wait for that `1 second`.
Isn't something like `100 milliseconds` enough for this mock unit test?
Don't we need to verify yet how many times that `mockConnection.createChannel()` is called because of retry enabled?

Reply:
You're absolutely right; this was an oversight on my part. I have made the modifications. Could you please confirm them? Thank you.